### PR TITLE
Report [GEAR-543]

### PR DESCRIPTION
**Pull request by dpark6060**
Added a report generation feature.

Report is printed at the end of the log
Report is also appended to the session as a .csv file with a timestamp in the file name.

Successful run:
https://ss.ce.flywheel.io/#/jobslog/job/5f4434fc7638810401588f90
On this session:
https://ss.ce.flywheel.io/#/projects/5e9f2b362971c80104f876b2/sessions/5e9f89a72971c80140f876c0?tab=data

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -29,3 +29,7 @@ The [DICOM Toolkit](https://support.dcmtk.org/docs/) (DCMTK) is a set of softwar
 1. *Acquire Inputs*. If the **file** input is not provided, all files of DICOM type in the session where the Gear is executed are downloaded and set as input for the next stage.
 2. *Prepare Inputs*. DICOMs packaged into archives (.tgz or .zip) are uncompressed. A private tag is then added to each DICOM file to be transmitted.
 3. *Transmit DICOMs*. The final stage transmits each DICOM file to the DICOM server indicated during Gear configuration.
+
+# Testing
+
+For information on gear testing, see the [testing readme](https://github.com/flywheel-apps/dicom-send/blob/master/TESTING.md).
```
Comment (lmperry) [inline README.md:35]:
```suggestion
For information on gear testing, see the [testing readme](TESTING.md).
```

File: manifest.json [hunk 2]
```diff
@@ -2,7 +2,7 @@
   "name": "dicom-send",
   "label": "DCMTK: DICOM Send",
   "description": "DICOM Send utilizes DCMTK's storescu to send DICOM data from a Flywheel instance to a destination DICOM server, hosted externally. This Gear supports the transmission of individual DICOM files and archives, as well as the transmission of an entire session when a specific input is not provided. Note that a private tag is added to each DICOM file to be transmitted (Flywheel:DICOM Send, at group 0x0021). Importantly, the external DICOM server must be reachable from the engine host of the Flywheel instance.",
-  "version": "1.0.0",
+  "version": "1.0.1",
```
Comment (lmperry) [inline manifest.json:5]:
```suggestion
  "version": "1.1.0",
```
For BF patch increment is good, but for feature addition, I think we can bump minor version.

File: manifest.json [hunk 3]
```diff
@@ -47,7 +47,7 @@
   "custom": {
       "gear-builder": {
           "category": "converter",
-          "image": "flywheel/dicom-send:1.0.0"
+          "image": "flywheel/dicom-send:1.0.1"
```
Comment (lmperry) [inline manifest.json:50]:
```suggestion
          "image": "flywheel/dicom-send:1.1.1"
```

File: utils/report_generator.py [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+    ses = fw.get_session(ses_id)
+    new_name = ses.label
+    
+    if acq_id is not None:
+        acq = fw.get_acquisition(acq_id)
+        new_name = f"{new_name}_{acq.label}"
+    
+    timestamp = datetime.now()
+    
+    new_name = f"{new_name}_{timestamp.strftime('%Y-%m-%d_%H:%M:%S')}.csv"
```
Comment (lmperry) [inline utils/report_generator.py:190]: Let's include the gear name here
```suggestion
    new_name = f"dicom-send_report-{new_name}_{timestamp.strftime('%Y-%m-%d_%H:%M:%S')}.csv"
```
I would be tempted to also include the job id, but I can't recall how readily accessible that is.

File: utils/report_generator.py [hunk 5]
```diff
@@ -0,0 +1,197 @@
...
+        csv_writer.writerow(list_of_elem)
+
+
+def initialize_report(file_name):
+    """Initializes a report .csv file with a specified set of headers.
+    
+    Args:
+        file_name (pathlib.PosixPath): The absolute path to the new .csv file 
+
+    Returns:
```
Comment (margaretmahan) [inline utils/report_generator.py:32]: I sometimes use the Returns section to say None if it returns None, but then explain what it does to the output if something happens. Or just delete this section if not used.

File: utils/report_generator.py [hunk 6]
```diff
@@ -0,0 +1,197 @@
...
+        print_string = 'REPORT_SUMMARY:\n'
+        for row in csv_f:
+            print_string+=format_string.format(*row)
+        
+        log.info(print_string)
+
+
+def upload_report(fw, ses_id, acq_id=None,
+                  report_file=Path('/flywheel/v0/report/dicom-send-report.csv')):
+    """Upload a target report .csv file to a session as an attachment
```
Comment (margaretmahan) [inline utils/report_generator.py:166]: NIT: If the arguments of a function need to be split for line length, I find it is easier to read if each argument is on a newline.

File: utils/report_generator.py (hunk 5 above)
Comment (margaretmahan) [reply utils/report_generator.py:32]: There are a few places this occurs in this script. Maybe clean up the docstrings with the Returns section completed or just left out.

File: utils/report_generator.py (hunk 6 above)
Comment (margaretmahan) [reply utils/report_generator.py:166]: Again, a few places this occurs for functions in this script.

File: utils/report_generator.py [hunk 7]
```diff
@@ -0,0 +1,197 @@
...
+    
+    Renames the report file based on the session and date.  If a single acquisition was specified
+    for export, the acquisition label is included too.
+    
+    Args:
+        fw (flywheel.Client()): the flywheel client for the current session
+        acq_id (str): the flywheel ID of the parent acquisition of the target file
+        ses_id (str): the flywheel ID of the parent session of the target file
+        report_file (str): the name of the target file
```
Comment (margaretmahan) [inline utils/report_generator.py:175]: Type posix Path or str?

File: utils/report_generator.py (hunk 5 above)
Comment (dpark6060) [reply utils/report_generator.py:32]: added "None" to be consistent with you

File: manifest.json (hunk 3 above)
Comment (dpark6060) [reply manifest.json:50]: should be 1.1.0, will change in next push

**Pull request merged**